pandas/io/formats/style.py

```diff
@@ -340,6 +343,319 @@ def to_excel(
             engine=engine,
         )
 
+    def to_latex(
+        self,
+        buf: FilePathOrBuffer[str] | None = None,
+        column_format: str | None = None,
+        position: str | None = None,
+        position_float: str | None = None,
+        hrules: bool = False,
+        label: str | None = None,
+        caption: str | None = None,
+        sparsify: bool | None = None,
+        multirow_align: str = "c",
+        multicol_align: str = "r",
+        siunitx: bool = False,
+        encoding: str | None = None,
+    ):
+        r"""
+        Write Styler to a file, buffer or string in LaTeX format.
+
+        .. versionadded:: TODO
+
+        Parameters
+        ----------
+        buf : str, Path, or StringIO-like, optional, default None
+            Buffer to write to. If ``None``, the output is returned as a string.
+        column_format : str, optional
+            The LaTeX column specification placed in location:
+
+            \\begin{tabular}{<column_format>}
+
+            Defaults to 'l' for index and
+            non-numeric data columns, otherwise 'r' (or 'S' if using the
+            {siunitx} package).
+        position : str, optional
+            The LaTeX positional argument (e.g. 'h!') for tables, placed in location:
+
+            \\begin{table}[<position>]
+        position_float : {"centering", "raggedleft", "raggedright"}, optional
+            The LaTeX float command placed in location:
+
+            \\begin{table}[<position>]
+
+            \\<position_float>
+        hrules : bool, default False
+            Set to `True` to add \\toprule, \\midrule and \\bottomrule from the
+            {booktabs} LaTeX package.
+        label : str, optional
+            The LaTeX label included as: \\label{<label>}.
+            This is used with \\ref{<label>} in the main .tex file.
+        caption : str, optional
```

Comment:
are these consistent (ordering & naming) with .to_latex()?

Reply:
naming: yes (with exceptions on multiindex formatting). ordering: not really.

There is sufficient change to the input arguments (see OP above), to warrant some re-ordering, for example all the formatting options are gone, and grouping the all arguments relating to multiindex sparsification seemed sensible.

Reply:
other than buf, the other parameters maybe could be keyword only.

Reply:
> other than buf, the other parameters maybe could be keyword only.

I agree.

Reply:
Thoughts on making keyword only @attack68?

Reply:
yes i definitely agree with this, it makes adding new kwargs later (and inserting them into certain places, i.e. kwarg groups much better).

As I understand PEP3102 it simply involves adding `, *,` before the proposed kwargs to indicate this.

Reply:
yeahlet's make keyword only